### PR TITLE
Mail service for async jobs

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR solves #1215 

* `Mail` service sends single emails directly or using a mailer class, driven by input `$payload` array
* a dummy `UserMailer` has been created as first mailer

### CODE REVIEW

File: plugins/BEdita/Core/src/Service/Mail.php [hunk 1]
```diff
@@ -14,21 +14,113 @@
...
+     *  * 'to' - Email recipient, mandatory if no 'mailer' is specified
+     *  * 'subject' - Email subject, mandatory if no 'mailer' is specified
+     *  * 'message' - Message body, mandatory if no 'mailer' is specified
+     *
+     * In $options array an alternative 'profile' name may be set.
+     *
+     * @param array $payload Input data for this email job.
+     * @param array $options Options for running this job.
+     * @return bool True on success, false on failure
      */
```
Comment (batopa) [inline plugins/BEdita/Core/src/Service/Mail.php]: Specify `@throws` in doc block as `* @throws \LogicException When bla bla`

File: plugins/BEdita/Core/src/Service/Mail.php [hunk 2]
```diff
@@ -14,21 +14,113 @@
...
+     */
+    protected $Email;
+
+    /**
+     * Send a single email directly or using a mailer class.
+     *
+     * $payload array data specify mail action and input data.
+     * Possible keys are:
+     *  * 'profile' - Email configuration profile to use (default is 'default'), in profile sender and from fields are
+     *  * 'mailer' - Custom mailer class to use, if no namespace is spesified a class with same name is searched
```
Comment (batopa) [inline plugins/BEdita/Core/src/Service/Mail.php]: 'spesified' typo

File: plugins/BEdita/Core/src/Service/Mail.php [hunk 3]
```diff
@@ -14,21 +14,113 @@
...
+
+    /**
+     * Send mail using mailer.
+     * If a plugin syntax is used for $name mailer is loaded directly.
+     * Otherwise a matching mailer is searched in plugins and core.
+     *
+     * @param string $name Mailer name
+     * @param string $action Mailer action
+     * @param array $options Mailer options
+     * @return bool True on success, false on failure
```
Comment (batopa) [inline plugins/BEdita/Core/src/Service/Mail.php:98]: `@throws` as above

**Review by batopa — COMMENTED** [review]
Minor changes in doc blocks

File: plugins/BEdita/Core/src/Service/Mail.php (hunk 2 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Service/Mail.php]: That's the `bolognaise` accent! 

Doc comment changed, should be more clear now .. not just for the typo

File: plugins/BEdita/Core/src/Service/Mail.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Service/Mail.php]: done

File: plugins/BEdita/Core/src/Service/Mail.php (hunk 3 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Service/Mail.php:98]: done

**Pull request merged**